pkg/service/schema.go

```diff
@@ -0,0 +1,94 @@
+package service
+
+// ParameterSchemas is a set of optional JSONSchemas that describe
+// the expected parameters for creation and update of instances and
+// creation of bindings.
+type ParameterSchemas struct {
+	ServiceInstances *instancesSchema `json:"service_instance,omitempty"`
+	ServiceBindings  *BindingsSchema  `json:"service_binding,omitempty"`
+}
+
+type instancesSchema struct {
+	Create *ProvisioningParametersSchema `json:"create,omitempty"`
+	Update *UpdatingParametersSchema     `json:"update,omitempty"`
+}
+
+// ProvisioningParametersSchema represents the schema for any parameters that
+// might be needed for provisioning a service instance
+type ProvisioningParametersSchema struct {
+	Parameters *ParametersSchema `json:"parameters,omitempty"`
+}
+
+// UpdatingParametersSchema represents the schema for any parameters that
+// might be needed for updating a service instance
+type UpdatingParametersSchema struct {
+	Parameters *ParametersSchema `json:"parameters,omitempty"`
+}
+
+// BindingsSchema represents a plan's schemas for the parameters
+// accepted for binding creation.
+type BindingsSchema struct {
+	Create *BindingSchema `json:"create,omitempty"`
+}
+
+// BindingSchema represents the schema for any parameters that
+// might be needed for creating a binding
+type BindingSchema struct {
+	Parameters *ParametersSchema `json:"parameters,omitempty"`
+}
+
+// ParametersSchema represents the individual schema for a service
+type ParametersSchema struct {
+	Schema     string               `json:"$schema"`
+	Type       string               `json:"type"`
+	Properties map[string]Parameter `json:"properties,omitempty"`
+	Required   []string             `json:"required,omitempty"`
+}
+
+// Parameter represents the individual schema for a given parameter
+type Parameter struct {
+	Type        string                 `json:"type"`
+	Description string                 `json:"description,omitempty"`
+	Default     interface{}            `json:"default,omitempty"`
+	Items       interface{}            `json:"items,omitempty"`
+	Properties  map[string]interface{} `json:"properties,omitempty"`
```

Comment:
Wondering if the type of this field should be `map[string]* Parameter`. (Or `map[string]* ParameterSchema` if that type is indeed renamed as suggested in a previous comment.)